packages/labs/analyzer/src/lib/types.ts

```diff
@@ -250,43 +259,57 @@ const getRootName = (
 let contents = '';
 let contentsId = 0;
 
-const service = ts.createLanguageService(
-  {
-    getScriptFileNames: () => ['contents.ts'],
-    getScriptVersion: () => contentsId.toString(),
-    getScriptSnapshot: () => ts.ScriptSnapshot.fromString(contents),
-    getCurrentDirectory: () => '',
-    getCompilationSettings: () => ({include: ['contents.ts']}),
-    getDefaultLibFileName: (options) => ts.getDefaultLibFilePath(options),
-    fileExists: () => true,
-    readFile: () => undefined,
-    readDirectory: () => [],
-    directoryExists: () => true,
-    getDirectories: () => [],
-  },
-  ts.createDocumentRegistry()
-);
+const services = new WeakMap<TypeScript, ts.LanguageService>();
+const getService = (typescript: TypeScript) => {
+  let service = services.get(typescript);
+  if (service === undefined) {
+    service = typescript.createLanguageService(
+      {
+        getScriptFileNames: () => ['contents.ts'],
+        getScriptVersion: () => contentsId.toString(),
+        getScriptSnapshot: () => typescript.ScriptSnapshot.fromString(contents),
+        getCurrentDirectory: () => '',
+        getCompilationSettings: () => ({include: ['contents.ts']}),
+        getDefaultLibFileName: (options) =>
+          typescript.getDefaultLibFilePath(options),
+        fileExists: () => true,
+        readFile: () => undefined,
+        readDirectory: () => [],
+        directoryExists: () => true,
+        getDirectories: () => [],
+      },
+      typescript.createDocumentRegistry()
+    );
+    services.set(typescript, service);
+  }
+  return service;
+};
 
 /**
  * Uses the stub LSH above to parse a type string and return a syntax-only
  * TypeNode (there will be no valid symbol information retrievable directly from
  * these nodes).
  */
-const parseType = (typeString: string): ts.TypeNode | undefined => {
+// TODO (justinfagnani): This would probably be a bit cleaner as an instance
+// method on Analyzer now.
```

Comment:
There will never be an easier time to make breaking changes than now

Reply:
This wouldn't be a breaking change, and would add public API, so I wanted to be conservative.

There are a lot of potential cleanups I saw while in here, I also needed to limit the changes to just the relevant ones to keep things reviewable.